pages/lessons/[lesson]/[slug].tsx

```diff
@@ -131,30 +125,42 @@ const Lesson: React.FC<LessonProps> = ({
 export default Lesson
 
 export const getStaticPaths = async () => {
-  const files = fs.readdirSync(path.join('lessons'))
-  const paths = files.map((filename) => ({
+  const trackDir = path.join('lessons')
```

Comment:
Hey Phil, I'd recommend moving all the "read and process mdx files" (from here and other files like Getting Started index page and Lesson index page) into its own module with methods like `getAllLessons()`, `getAllFundamentals()` etc. Then the React components get much simpler and less code duplication across files.

Reply:
Method names were just examples. Maybe something more dynamic like `getContentSection('fundamentals')` and `getAllContentSectionTypes()` versus a method per type of content, etc.